src/gflownet/tasks/seh_frag.py

```diff
@@ -52,25 +58,38 @@ def _load_task_models(self):
         model, self.device = self._wrap_model(model)
         return {'seh': model}
 
-    def sample_conditional_information(self, n):
+    def sample_conditional_information(self, n: int) -> Dict[str, Tensor]:
         beta = None
-        if self.temperature_sample_dist == 'gamma':
-            loc, scale = self.temperature_dist_params
-            beta = self.rng.gamma(loc, scale, n).astype(np.float32)
-            upper_bound = stats.gamma.ppf(0.95, loc, scale=scale)
-        elif self.temperature_sample_dist == 'uniform':
-            beta = self.rng.uniform(*self.temperature_dist_params, n).astype(np.float32)
-            upper_bound = self.temperature_dist_params[1]
-        elif self.temperature_sample_dist == 'beta':
-            beta = self.rng.beta(*self.temperature_dist_params, n).astype(np.float32)
-            upper_bound = 1
-        beta_enc = thermometer(torch.tensor(beta), 32, 0, upper_bound)  # TODO: hyperparameters
-        return {'beta': torch.tensor(beta), 'encoding': beta_enc}
-
-    def cond_info_to_reward(self, cond_info: Dict[str, Tensor], flat_reward: FlatRewards) -> RewardScalar:
+        if self.temperature_sample_dist == 'constant':
+            assert type(self.temperature_dist_params) is float
+            beta = np.array(self.temperature_dist_params).repeat(n).astype(np.float32)
+            beta_enc = torch.zeros((n, self.num_thermometer_dim))
+        else:
+            if self.temperature_sample_dist == 'gamma':
+                loc, scale = self.temperature_dist_params
+                beta = self.rng.gamma(loc, scale, n).astype(np.float32)
+                upper_bound = stats.gamma.ppf(0.95, loc, scale=scale)
+            elif self.temperature_sample_dist == 'uniform':
+                beta = self.rng.uniform(*self.temperature_dist_params, n).astype(np.float32)
+                upper_bound = self.temperature_dist_params[1]
+            elif self.temperature_sample_dist == 'loguniform':
+                beta = np.exp(self.rng.uniform(*np.log(self.temperature_dist_params), n).astype(np.float32))
```

Comment:
mypy is having trouble following the `*` operator here unfortunately, because it cannot determine the length of the output of `np.log(..)` and finds no match for a two-argument `uniform(Number, Integer)` call. A fix is to declare the uniform bounds outside e.g.:
```
low, high = np.log(self.temperature_dist_params)
beta = np.exp(self.rng.uniform(low, high, n).astype(np.float32))
```

Reply:
Fixed [here](https://github.com/recursionpharma/gflownet/commit/1a952537df52ebd462f3d6e2d0942a8277c73868)